controllers/clusteroperator_test.go

```diff
@@ -2,41 +2,36 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"strings"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
+	configv1 "github.com/openshift/api/config/v1"
+	osconfigv1 "github.com/openshift/api/config/v1"
+	fakeconfigclientset "github.com/openshift/client-go/config/clientset/versioned/fake"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
 	"k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	osconfigv1 "github.com/openshift/api/config/v1"
-	fakeconfigclientset "github.com/openshift/client-go/config/clientset/versioned/fake"
 	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
 	provisioning "github.com/openshift/cluster-baremetal-operator/provisioning"
-	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
 )
 
 func TestUpdateCOStatusDisabled(t *testing.T) {
 	tCases := []struct {
 		name               string
 		expectedConditions []osconfigv1.ClusterOperatorStatusCondition
-		expected           bool
 	}{
-		{
-			name: "Incorrect Condition",
-			expectedConditions: []osconfigv1.ClusterOperatorStatusCondition{
-				setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, "", ""),
-				setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, "", ""),
-			},
-			expected: false,
-		},
 		{
 			name: "Correct Condition",
 			expectedConditions: []osconfigv1.ClusterOperatorStatusCondition{
+				setStatusCondition(osconfigv1.OperatorDegraded, osconfigv1.ConditionFalse, "", ""),
 				setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, "", ""),
-				setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, "", ""),
+				setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, "UnsupportedPlatform", "Operator is non functional"),
```

Comment:
If this "disabled" behavior is meant to be part of the API of an operator, should we add it to the same package where the other condition constants are defined?

Reply:
The other states are defined here; https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go#L141. There was a discussion earlier with @deads2k to add support for the "Disabled" state there and the answer then was a No. At this time, there are only a very small number of operators that use this state, that might have something to do with that decision.